src/vs/workbench/parts/terminal/common/terminal.ts

```diff
@@ -0,0 +1,53 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+'use strict';
+
+import {TPromise} from 'vs/base/common/winjs.base';
+import {createDecorator, ServiceIdentifier} from 'vs/platform/instantiation/common/instantiation';
+import path = require('path');
+import platform = require('vs/base/common/platform');
+
+export const TERMINAL_PANEL_ID = 'workbench.panel.terminal';
+
+export const TERMINAL_SERVICE_ID = 'terminalService';
+
+export const TERMINAL_DEFAULT_SHELL_UNIX_LIKE = process.env.SHELL || 'sh';
+export const TERMINAL_DEFAULT_SHELL_WINDOWS = platform.isWindows ? path.resolve(process.env.SystemRoot, 'System32', 'WindowsPowerShell', 'v1.0', 'powershell.exe') : '';
+
+export var ITerminalService = createDecorator<ITerminalService>(TERMINAL_SERVICE_ID);
+
+export interface ITerminalConfiguration {
+	integratedTerminal: {
+		shell: {
+			unixLike: string,
+			windows: string
+		},
+		fontFamily: string,
+		ansiColors: {
+			black: string,
+			red: string,
+			green: string,
+			yellow: string,
+			blue: string,
+			magenta: string,
+			cyan: string,
+			white: string,
+			brightBlack: string,
+			brightRed: string,
+			brightGreen: string,
+			brightYellow: string,
+			brightBlue: string,
+			brightMagenta: string,
+			brightCyan: string,
+			brightWhite: string,
+		}
+	};
+}
+
+export interface ITerminalService {
```

Comment:
I figured this could be useful later on when things like multiple terminals and switching between them are added.